### PR TITLE
Refactor EXP-3953 [v121] Move nimbus-fml.sh to application-services

### PR DESCRIPTION
**Pull request by jhugman**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/EXP-3953)

## :bulb: Description

This moves the `nimbus-fml.sh` out of Firefox for iOS and into Application Services. The [app services PR][app-services] adds a bootstrap script which can be curled and executed as part of this project's bootstrap. This is now documented in [`experimenter.info` docs][docs].

This has been a long held, low priority goal. The pricipitating events:

- the Nimbus team is asking engineers [to add more metadata to their Nimbus powered features](https://experimenter.info/fml/feature-metadata).
  - we have added a `validate` command to the `nimbus-fml` command line
  - the `validate` command checks for the relevant metadata. These warnings are emitted in the Build console logs.
  - this PR adds metadata for the `messaging` feature.
- the `generate-experimenter` command is now being run by experimenter, so is no longer needed to be commited into the repository.
  - this PR removes the `.experimenter.yaml` file

Once merged, engineers will need to re-run the `bootstrap.sh` script on their machines to re-download the `nimbus-fml.sh` script (it is removed, and added to `.gitignore` in this PR).

This should otherwise provide no extra functionality or tests.

[app-services]: https://github.com/mozilla/application-services/pull/5877
[docs]: https://github.com/mozilla/experimenter-docs/pull/519

Commits:

- Remove nimbus-fml.sh in favour of bootstrapped from AS copy
- Remove .experimenter.yaml
- Add feature metadata for messaging feature
- Trigger FML generate on every build

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [X] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -37,5 +37,9 @@ if [ "$1" == "--importLocales" ]; then
   exit 0
 fi
 
+# Download the nimbus-fml.sh script from application-services.
+NIMBUS_FML_FILE=./nimbus.fml.yaml
```
Comment (jhugman) [inline bootstrap.sh:41]: Should you ever move the `nimbus.fml.yaml` file, you should change this.

File: .experimenter.yaml [hunk 2]
```diff
@@ -1,228 +0,0 @@
----
-contextual-hint-feature:
```
Comment (adudenamedruby) [inline .experimenter.yaml:2]: How come the entire '.experimenter.yaml' file gets removed?

File: nimbus-features/messagingFeature.yaml [hunk 3]
```diff
@@ -2,6 +2,24 @@ features:
...
+      - name: User documentation
+        url:  https://experimenter.info/messaging/mobile-messaging
+    contacts:
+      # Nimbus team
+      - jhugman@mozilla.com
+      - brennie@mozilla.com
+      # iOS team
+      - gbuciu@mozilla.com
+      - wtiechmann@mozilla.com
+      - npatel@mozilla.com
```
Comment (lmarceau) [inline nimbus-features/messagingFeature.yaml:16]: this person doesn't work at Mozilla anymore, and there's a typo in Winnie's email
```suggestion
      - wteichmann@mozilla.com
```

Comment (jhugman) [reply nimbus-features/messagingFeature.yaml]: Gosh darn it! Thanks!

File: .experimenter.yaml (hunk 2 above)
Comment (jhugman) [reply .experimenter.yaml:2]: > the `generate-experimenter` command is now being run by experimenter, so is no longer needed to be commited into this repository.

It's only function was to be ingested by experimenter, and now experimenter can ingest the FML directly.

**Comment by isabelrios** [comment]
@jhugman let me re-run this PR so that it is unblocked to be merged. Context: Bitrise is red because the UI Smoketest run isolated, not as part of the pipeline and failed because it misses the dependencies.

**Comment by jhugman** [comment]
Thank you @isabelrios !

File: .experimenter.yaml (hunk 2 above)
Comment (adudenamedruby) [reply .experimenter.yaml:2]: Amazing!

**Pull request merged**